File: DSA/DSA08002 CẤU TRÚC DỮ LIỆU HÀNG ĐỢI 2.cpp
#include <bits/stdc++.h>
using namespace std;

void submit()
{
    int n;
    cin >> n;
    cin.ignore();
    queue<int> A;

    for (int i = 0; i < n; i++)
    {
        string s;
        getline(cin, s);

        stringstream ss(s);

        string temp;
        ss >> temp;

        if (temp == "PUSH")
        {
            int idk;
            ss >> idk;
            A.push(idk);
        }
        else if (temp == "POP")
        {
            if (!A.empty()) A.pop();
        }
        else
        {
            !A.empty() ? cout << A.front() : cout << "NONE";
            cout << endl;
        }
    }
}

int main()
{
    ios::sync_with_stdio(false);                                    // Speed up input/output
    cin.tie(nullptr);
    cout.tie(nullptr);

    submit();

    return 0;
}

File: DSA_python/DSA08002 CẤU TRÚC DỮ LIỆU HÀNG ĐỢI 2.py
def submit():
    n = int(input())
    A = list()

    for i in range(n):
        s = input().split()

        if s[0] == "PUSH":
            A.append(s[1])
        elif s[0] == "POP":
            if len(A) != 0: A.remove(A[0])
        else:
            print(A[0] if len(A) != 0 else "NONE")


submit()
